harness/ts/harness.ts

```diff
@@ -0,0 +1,138 @@
+import { Connection, WorkflowClient, WorkflowHandleWithRunId } from '@temporalio/client';
+import { ActivityInterface, Workflow } from '@temporalio/common';
+import { Worker } from '@temporalio/worker';
+import { promises as fs } from 'fs';
+import * as path from 'path';
+import { v4 as uuidv4 } from 'uuid';
+
+export interface FeatureOptions<W extends Workflow, A extends ActivityInterface> {
+  // Defaults to import(workflowsPath).workflow
+  workflow?: W;
+  // Defaults to no activities
+  activities?: A;
+  workflowsPath?: string;
+  execute?: (runner: Runner<W, A>) => Promise<WorkflowHandleWithRunId<W>>;
+  checkResult?: (runner: Runner<W, A>, handle: WorkflowHandleWithRunId<W>) => Promise<void>;
+  checkHistory?: (runner: Runner<W, A>, handle: WorkflowHandleWithRunId<W>) => Promise<void>;
+}
+
+export class Feature<W extends Workflow, A extends ActivityInterface> {
```

Comment:
nit: Technically you don't even need a class for this

Reply:
As an interface, it was not properly inferring the types. Can you describe what alternative you were thinking and how that would look inside the feature file?